### PR TITLE
[23890] Add typography components

### PR DESCRIPTION
**Pull request by renatofreire**
# What

We did Typography components: Text and Heading.

# Why

Hackathon purposes. 

# How

Pair programming @tobiasbu 

# Sample

Run tests or the storybook (search by Heading and Texts)

### CODE REVIEW

File: packages/components/src/Heading/__tests__/index.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+import Heading from '../index';
+import { theme } from '@magnetis/astro-galaxy-core';
+
+describe('Heading', () => {
+  it('should render', () => {
```
Comment (brunobc182) [inline packages/components/src/Heading/__tests__/index.test.js]: I think `renders correctly` fits better in this description.

File: packages/components/src/Text/__tests__/index.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { theme } from '@magnetis/astro-galaxy-core';
+
+import Text from '../index';
+
+describe('Text', () => {
+  it('should render', () => {
```
Comment (brunobc182) [inline packages/components/src/Text/__tests__/index.test.js]: Same as `Heading` comment.

File: packages/core/src/themes/control.js [hunk 3]
```diff
@@ -149,6 +149,20 @@ const theme = {
...
+      verySmall: `${baseTheme.fontSizes[0]}px`,
+      small: `${baseTheme.fontSizes[1]}px`,
+      medium: `${baseTheme.fontSizes[2]}px`,
+      large: `${baseTheme.fontSizes[3]}px`,
+    },
+    headings: {
+      small: `${baseTheme.fontSizes[4]}px`,
+      medium: `${baseTheme.fontSizes[5]}px`,
+      large: `${baseTheme.fontSizes[6]}px`,
+      huge: `${baseTheme.fontSizes[7]}px`,
```
Comment (AgtLucas) [inline packages/core/src/themes/control.js]: Captain Raymold Holt saying, I've never been happier

File: packages/components/src/Heading/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+`;
+
+Heading.displayName = 'Heading';
+
+Heading.defaultProps = {
+  color: 'moon900',
+  fontSize: 'large',
+};
+
+Heading.propTypes = {};
```
Comment (AgtLucas) [inline packages/components/src/Heading/index.js]: Let's add proptypes here now, or we can do this after the Astrothon.

Your choice.

File: packages/components/src/Text/index.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import styled from 'styled-components';
+import { color, space, border, layout, buttonStyle, typography } from 'styled-system';
+import { font } from '@magnetis/astro-galaxy-core';
+
+const Text = styled.p`
+  color: ${props => props.theme.colors[props.color]};
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
```
Comment (AgtLucas) [inline packages/components/src/Text/index.js]: All texts should have these properties?

If so, let's move them to `packages/core`, shall we?

File: packages/components/src/Text/index.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+Text.propTypes = {
+  ...color.propTypes,
+  ...space.propTypes,
+  ...border.propTypes,
+  ...layout.propTypes,
+  ...buttonStyle.propTypes,
+  ...typography.propTypes,
+};
```
Comment (AgtLucas) [inline packages/components/src/Text/index.js]: If we gonna use these props, we shall pass them to the component itself. For instance: https://github.com/magnetis/astro-galaxy/blob/master/packages/components/src/Button/index.js#L72-L80

File: packages/core/src/theme-provider/index.js [hunk 7]
```diff
@@ -77,6 +77,36 @@ const GlobalStyle = createGlobalStyle`
...
+  font-weight: 700;
+  src:
+    local('Poppins Bold'),
+    local('Poppins-Bold'),
+    url(https://fonts.gstatic.com/s/poppins/v6/pxiByp8kv8JHgFVrLCz7Z1xlFd2JQEk.woff2) format('woff2'),
+    url(https://fonts.gstatic.com/s/poppins/v6/pxiByp8kv8JHgFVrLCz7Z1xlFd2JQEk.woff) format('woff'),
+    url(https://fonts.gstatic.com/s/poppins/v6/pxiByp8kv8JHgFVrLCz7Z1xlFd2JQEk.eot?#) format('eot');
+  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
+}
+
```
Comment (AgtLucas) [inline packages/core/src/theme-provider/index.js]: Oh damn, we now have weight 700. 😮 

Let's just format this better.

File: packages/components/src/Heading/__tests__/index.test.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  describe('when heading "color" prop is set', () => {
+    it('renders with "mars600"', () => {
+      const json = rendererCreateWithTheme(<Heading color="mars600" />).toJSON();
+      expect(json).toHaveStyleRule('color', theme.colors.mars600);
+    });
+  });
+
+  describe('when heading "fontSize" prop is set', () => {
+    it('renders with "small" font size ', () => {
+      const json = rendererCreateWithTheme(<Heading fontSize="small" />).toJSON();
```
Comment (AgtLucas) [inline packages/components/src/Heading/__tests__/index.test.js]: Let's add actual text for all these `Heading` and `Text` tests.

```js
<Heading fontSize="small">minha carteira</Heading>
```

File: packages/components/src/Heading/index.js (hunk 4 above)
Comment (renatofreire) [reply packages/components/src/Heading/index.js]: We did Agent Lucas.

File: packages/components/src/Text/index.js (hunk 5 above)
Comment (renatofreire) [reply packages/components/src/Text/index.js]: No. Astro does not have this for *all* textual CSS. Only headings and paragraphs.

**Pull request merged**